components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementPolicyAdminService.java

```diff
@@ -369,9 +370,6 @@ public void removePolicy(String policyId, boolean dePromote) throws EntitlementE
         }
         handleStatus(EntitlementConstants.StatusTypes.DELETE_POLICY, oldPolicy, true, null);
 
-        //remove versions
-        EntitlementAdminEngine.getInstance().getVersionManager().deletePolicy(policyId);
```

Comment:
Removing policy version is now moved to `policyAdmin.removePolicy(policyId)`. Validate whether this is ok.

Reply:
This should be ok, since `PolicyDAO.removePolicy()` only get called from here.